survival_lib.py

```
@@ -176,63 +217,153 @@ def fit_km_logrank(df, duration_col, event_col, group_col):
             
-            if ci_exists and kmf.confidence_interval_.shape[1] >= 2:
-                ci_lower = kmf.confidence_interval_.iloc[:, 0].values
-                ci_upper = kmf.confidence_interval_.iloc[:, 1].values
-                ci_times = kmf.confidence_interval_.index.values
+            if len(df_g) > 0:
+                kmf = KaplanMeierFitter()
                 
-                rgba_color = _hex_to_rgba(colors[i % len(colors)], 0.2)
+                # Robust fit
+                CONNECTION_HANDLER.retry_with_backoff(
+                    lambda: kmf.fit(df_g[duration_col], df_g[event_col], label=label)
```

Ruff (0.14.10)

Function definition does not bind loop variable `kmf`

(B023)